Docker builds

davidthor opened the pull request:
This PR includes the code necessary to satisfy #10 and build docker images to be published.

ryan-cahill commented on Dockerfile-Base:
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:16.04
```
Any reason we're not using 18.04?

ryan-cahill commented on Dockerfile-Base:
```diff
@@ -0,0 +1,23 @@
+  apt-get install -y nodejs
+
+# Map python3 to python
+RUN sudo rm /usr/bin/python && \
```
Why does sudo need to be installed and run here? When you're running a container you should already be root by default

ryan-cahill commented on src/commands/build.ts:
```diff
@@ -0,0 +1,63 @@
+
+  async buildImage(service_path: string) {
+    const {flags} = this.parse(Build);
+    const service_config = ServiceConfig.loadFromPath(service_path);
```
Should there be a check after this to make sure that a service config was actually found?

davidthor replied:
`loadFromPath` already throws an error and triggers an exit when the path doesn't exist.

davidthor commented:
Updated based on feedback and added tests for the new build command. @ryan-cahill the overall tests will fail for unrelated, already documented reasons, but can you verify the integrity of the tests?

ryan-cahill commented on Dockerfile-Base:
```diff
@@ -0,0 +1,23 @@
+# Install Node.js & Python
+RUN apt-get update && \
+  apt-get -y upgrade && \
+  apt-get install -y build-essential autoconf libtool pkg-config curl git sudo unzip python3-pip && \
```
Shouldn't need to install sudo here

ryan-cahill commented on Dockerfile-Base:
```diff
@@ -0,0 +1,23 @@
+RUN apt-get update && \
+  apt-get -y upgrade && \
+  apt-get install -y build-essential autoconf libtool pkg-config curl git sudo unzip python3-pip && \
+  pip3 install --upgrade pip click grpcio && \
```
This pip install and the node install should probably be in different run commands than the apt-get installs so that if one command fails, the others don't need to be re-run for no reason

davidthor replied:
That creates more layers which increases the image size. Recommendations are to perform commands the way shown.

davidthor commented:
All tests pass locally os of merging. Closing.
